### PR TITLE
[http] Swap default to legacy codec with runtime override to new (no-op)

### PR DESCRIPTION
**Pull request by asraa**
Commit Message: Swap default codec to legacy codec implementation while codec error handling is improved. Backport some non-backported changes.
Risk Level: Low right now, legacy and codecs have not diverged
Testing: Integration tests now run by default with legacy, compile time options with new codecs. Unit tests are parametrized.
Release Notes: Updated
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by asraa** [comment]
The diff checker caused too many people's PRs to break format, so it got reverted https://github.com/envoyproxy/envoy/commit/6f17ae351ce006ea121abca3e1877c320f5530ca
(I think amongst merges was the real conflict... two codec merges would end up with conflicting diffs and break the diff checker).
Suggestions for how to do this cleanly? A big WARNING APPLY CHANGES TO LEGACY if format sees a change in codecs but no enforcement of a strict diff?

What's surprising is tests didn't pick up some of the functional detail changes.

**Comment by alyssawilk** [comment]
Ahhh, I missed that, so it was my bad.  how about a simpler check that if you edit one file you edit the other?  That at least would make sure folks touched both and the reviewer could make sure it roughly looked right.

**Comment by alyssawilk** [comment]
That suggestion orthogonal of the LGTM mind you :-P  Merge at your convenience!

**Comment by asraa** [comment]
Cool, will do in a follow-up!

**Comment by asraa** [comment]
@mattklein123 Could you take a quick look at this for cross-org stamp? (It should be a very quick review. Had to backport some changes to legacy, currently no diff exists between new/legacy's except namespace diffs).

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -44,7 +44,7 @@ New Features
...
 * grpc-json: support specifying `response_body` field in for `google.api.HttpBody` message.
 * http: added support for :ref:`%DOWNSTREAM_PEER_FINGERPRINT_1% <config_http_conn_man_headers_custom_request_headers>` as custom header.
-* http: introduced new HTTP/1 and HTTP/2 codec implementations that will remove the use of exceptions for control flow due to high risk factors and instead use error statuses. The old behavior is deprecated, but can be used during the removal period by setting the runtime feature `envoy.reloadable_features.new_codec_behavior` to false. The removal period will be one month.
+* http: introduced new HTTP/1 and HTTP/2 codec implementations that will remove the use of exceptions for control flow due to high risk factors and instead use error statuses. The old behavior is used by default, but the new codecs can be enabled for testing by setting the runtime feature `envoy.reloadable_features.new_codec_behavior` to false. The new codecs will be in development for one month, and then enabled by default while the old codecs are deprecated.
```
Comment (mattklein123) [inline docs/root/version_history/current.rst:47]:
```suggestion
* http: introduced new HTTP/1 and HTTP/2 codec implementations that will remove the use of exceptions for control flow due to high risk factors and instead use error statuses. The old behavior is used by default, but the new codecs can be enabled for testing by setting the runtime feature `envoy.reloadable_features.new_codec_behavior` to true. The new codecs will be in development for one month, and then enabled by default while the old codecs are deprecated.
```

?

Comment (asraa) [reply docs/root/version_history/current.rst]: Yes!!! Thanks for the catch. Done

**Pull request merged**